### ISSUE TITLE
Timeout web console live data layouts to something static to save TTI servers un-watched traffic

### ISSUE THREAD

**Issue by descartes** [TheThingsNetwork/lorawan-stack#5170]
#### Summary
Timeout a console window if it is showing any form of live data to some resting place - or pause the feed

#### Why do we need this?
So Chrome doesn't kill your servers & my machine when I leave too many tabs open.

This is particularly problematic (mainly for TTI) when I've got three or four machines with various tabs running ...

#### What is already there? What do you see now?
I'm trying to get in to the habit of clicking Applications at the top of the window, but it's down to me to remember.

#### What is missing? What do you want to see?
Perhaps an embedded timeout that can be adjusted on the user settings page and on the application page.

I'd be happy for it to be 60 minutes by default and be able to set it up to 12 hours (longer than a normal working day) on a per application basis.

#### How do you propose to implement/test/DIY this?
I think this is an @kschiffer thing

**Comment by descartes** [comment]
This strikes me as a win for all parties - I don't toast my workstations with lots of console tabs left open, you don't have the servers toasted serving console data to no one watching.

**Comment by kschiffer** [comment]
I agree that some solution is needed here. My initial idea was to automatically pause live data tabs that are not active but that might be a bit too aggressive. Probably, a timeout for unfocused tabs will be good. l don't know if we need any restrictions for focused live-data tabs. If a user wants to actively watch the live data for hours on end, they should be good to do so.

As far as memory usage goes, the Console applies a hardcap of events that it will display, which is currently set to 2000 and should be restrictive enough to prevent memory issues.

**Comment by descartes** [comment]
Unfocused tabs after a period of time are the real issue - as I walk away from my monitors I can see what's running live data which I usually only leave open to capture info. But as I have Safari & Chrome on my iMac and Chrome on my iWin (an iMac running Windows 10) and usually have ~10 windows with ~20 tabs on each, losing track of random tabs that are receiving events is oh so easy.

It's not a memory issue (24Gb can take a lot of tabs), it's when Chrome goes off on one for no apparent reason and superheats the machines. Closing any tabs with active code on them helps considerably. That said, sometimes 10,000 console lines would be useful!

I have tried running the CLI but occasionally that disconnects to pick up the details on an occasional glitch - whereas at least the web console reconnects automagically!

**Comment by kschiffer** [comment]
> That said, sometimes 10,000 console lines would be useful!

Yeah it's a thin edge here. I believe once we have proper filtering in place that will also solve many issues here since users can choose to not ingest any irrelevant events.

**Comment by descartes** [comment]
I have that as a standalone client side page since Feb '21 - but because it needs an API key, even a read only one, people get in a tangle thinking I'm gaining access to their bank account I don't share it as support is a mess.

**Comment by KrishnaIyer** [comment]
We have filtering in place and console events have a dedicate API.

**Comment by HeadBoffin** [comment]
> We have filtering in place and console events have a dedicate API.

This doesn't address the problem that is having multiple console tabs open in multiple browsers on multiple machines - which in my case has now grown to three machines + monitors on my desk, plus two laptops.

Surely this has an impact on the backend delivering events?
